lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1327,6 +1327,41 @@ void AtenSizeOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenSelectIntOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenSelectIntOp::fold(FoldAdaptor adaptor) {
+  auto self = dyn_cast_or_null<DenseElementsAttr>(adaptor.getSelf());
+  auto ty = cast<ValueTensorType>(getType());
```

Comment:
I see that CI failed.  Maybe you're caught in the same canonicalization loop that I was: eventually it will be castable to a ValueTensorType, but not the first time you get called

Reply:
I am surprised it fails (probably related to `BaseTensorType`) but either way, change to `dyn_cast` should fix.

Reply:
I had the same problem with index_select.  It turns out that the first time the folder is called, the tensor is still `!torch.tensor`, with no shape or size.  It seems the tensor is resolved to `!torch.vtensor` in several steps, so if you were to print out the type of the tensor at this point, you would see your print statement several times (once per call of your folder, of course) with different types, progressing toward a fully shaped and typed vtensor.